### PR TITLE
release: better-auth-telegram v2.0.0

### LINKED ISSUE #26: Miniapp auto login from hash params
### Problem

I’m not using any SDK for the miniapp (don’t really need it), but it is possible to get `initData` with a simple trick.

### Proposed Solution

```svelte
<script lang="ts">
	import { authClient } from "$lib/auth-client";
	import { browser } from "$app/environment";

	const session = authClient.useSession();

	let isInMiniapp = $state(false);
	if (browser && window.location.hash.length) {
		try {
			const params = new URLSearchParams(window.location.hash.slice(1));
			const initData = params.get("tgWebAppData");
			if (initData) {
				authClient.signInWithMiniApp(initData).then(({ data }) => {
					// hide logout button since it doesn’t really make sense:
					if (data) isInMiniapp = true;
				});
			}
		} catch {
			// some other hash? do nothing
		}
	}
</script>

{#if $session.data}
	<div>
		<span>{$session.data.user.name}</span>
		{#if !isInMiniapp}
			<button onclick={async () => await authClient.signOut()}>log out</button>
		{/if}
	</div>
{:else}
	<button onclick={async () => await authClient.signInWithTelegramOIDC()}>log in</button>
{/if}
```

Might try something like that in the `autoSignInFromMiniApp()` function if `window.Telegram` is not available.

### Alternatives Considered

_No response_

### Additional Context

https://docs.telegram-mini-apps.com/platform/launch-parameters

### PR DESCRIPTION
**Pull request by vcode-sh**
## What changed

- updates the supported Better Auth range to `>=1.6.22 <1.7.0`
- makes Telegram credentials flow-aware instead of crashing unrelated setups at startup
- adds Mini App auto-sign-in from the `tgWebAppData` hash when the SDK is not present
- updates Telegram OIDC claims and restricts JWT verification to supported, matching algorithms
- refreshes dependencies and documentation without changing existing OIDC account IDs behind users' backs

## Why 2.0.0

The Better Auth security floor drops support for 1.5 and older 1.6 releases. That is a real breaking requirement, so this is a real major release. No semver gymnastics.

We deliberately did not switch persisted OIDC identity from `sub` to Telegram's numeric `id`: doing that without a migration could create duplicate users. That work stays separate and gets the care identity migrations deserve.

## Validation

- `npm ci`
- `npm run prepublishOnly` — typecheck, 344 tests, ESM/CJS/DTS build
- `npm run lint`
- `npm run test:coverage` — 100% statements, lines, and functions; 98.72% branches
- `npm audit --omit=dev` — 0 vulnerabilities
- isolated typecheck, 344 tests, and build against Better Auth 1.6.22
- `npm pack --dry-run`

Closes #26 after the release is published.

## Summary by CodeRabbit

* **New Features**
  * Added Mini App sign-in fallback using data from the URL when Telegram WebApp data is unavailable.
  * Added support for additional Telegram OIDC claims, including names and phone verification status.
  * OIDC-only configurations can now omit widget-specific bot credentials.

* **Bug Fixes**
  * Missing credentials now produce actionable warnings during setup and clear errors when the affected flow is used.
  * Improved OIDC token validation supports RS256, ES256, and EdDSA while rejecting unsupported or mismatched algorithms.

* **Documentation**
  * Updated setup, migration, security, API, and troubleshooting guidance for version 2.0.0.

### CODE REVIEW

File: docs/api-reference.md [hunk 1]
```diff
@@ -309,7 +312,7 @@ if (result.data?.valid) {
 
 #### `autoSignInFromMiniApp(fetchOptions?)`
 
-The lazy developer's dream. Automatically grabs `initData` from `window.Telegram.WebApp.initData` and signs in. Only works inside a Telegram Mini App -- throws if you try it in a regular browser.
+Automatically grabs `initData` from `window.Telegram.WebApp.initData`, falling back to the `tgWebAppData` URL-fragment launch parameter, and signs in. Throws when neither source is available.
```
[the remark this answers is not in the document]
Comment (vcode-sh) [reply docs/api-reference.md:315]: Already covered in the release branch: the method now says it throws only when neither the SDK value nor tgWebAppData is available.

File: docs/configuration.md [hunk 2]
```diff
@@ -129,13 +130,17 @@ telegram({
 | Option | Type | Default | Description |
 |--------|------|---------|-------------|
 | `enabled` | `boolean` | `false` | Turn on Telegram OIDC. Injects a `telegram-oidc` provider into Better Auth's social login system via the `init` hook. |
+| `clientId` | `string` | bot ID from `botToken` | Client ID from BotFather Web Login. Required when `botToken` is omitted. |
+| `clientSecret` | `string` | `botToken` fallback | Client Secret from BotFather Web Login. Configure it explicitly; a bot token is not a valid OIDC client secret. |
```
[the remark this answers is not in the document]
Comment (vcode-sh) [reply docs/configuration.md]: Clarified in c4a0cdb. The implementation keeps the bot-token fallback for compatibility, while the docs now call out the BotFather Web Login Client Secret required by Telegram’s official OIDC setup.

File: docs/README.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ That's it. The rest is in the [Installation Guide](./installation.md) for people
 ## Features
 
 - **Login Widget** -- callback and redirect modes, because one way to authenticate was never enough
-- **OIDC (OpenID Connect)** -- standard OAuth 2.0 with PKCE via `oauth.telegram.org`. Phone numbers, RS256 JWTs, grown-up auth. Telegram finally joined the federation
+- **OIDC (OpenID Connect)** -- standard OAuth 2.0 with PKCE via `oauth.telegram.org`. Phone numbers and JWKS-verified JWTs, without the legacy widget callback
```
[the remark this answers is not in the document]
Comment (vcode-sh) [reply docs/README.md]: Fixed in c4a0cdb: phone numbers are now explicitly tied to requesting the phone scope.

File: src/oidc.ts [hunk 4]
```diff
@@ -38,7 +39,13 @@ const getTelegramPublicKey = async (kid: string) => {
...
   }
 
-  return await importJWK(jwk, jwk.alg);
+  if (jwk.alg !== algorithm) {
+    throw new Error(
+      `JWK algorithm does not match token algorithm ${algorithm}`
+    );
+  }
+
+  return await importJWK(jwk, algorithm);
```
[the remark this answers is not in the document]
Comment (vcode-sh) [reply src/oidc.ts]: Good catch. Fixed test-first in c4a0cdb: the new duplicate-kid regression test failed on the old lookup, and verification now selects by kid + alg. Full suite is green at 345 tests.

**Comment by vcode-sh** [comment]
Review follow-up is in c4a0cdb. The config response now has one shared type; it keeps botUsername as a string because the server endpoint always resolves it to a string (empty when not configured), and widget consumers still reject an empty value. I also fixed the real duplicate-kid JWKS edge case test-first and cleaned up the OIDC docs. Local release gate: lint, type-check, 345 tests, ESM/CJS/DTS build — all green.
